lib/mayaUsd/fileio/translators/translatorCamera.cpp

```diff
@@ -204,13 +208,173 @@ static bool _CreateAnimCurveForPlug(
     return true;
 }
 
+#if PXR_VERSION >= 2411
+static MFnAnimCurve::InfinityType _ConvertUsdExtrapolationTypeToMaya(TsExtrapMode usdExtrapolation)
+{
+    switch (usdExtrapolation) {
+    case TsExtrapLinear: return MFnAnimCurve::InfinityType::kLinear;
+    case TsExtrapLoopReset: return MFnAnimCurve::InfinityType::kCycle;
+    case TsExtrapLoopOscillate: return MFnAnimCurve::InfinityType::kOscillate;
+    case TsExtrapLoopRepeat: return MFnAnimCurve::InfinityType::kCycleRelative;
+    case TsExtrapHeld:
+    default: return MFnAnimCurve::InfinityType::kConstant;
+    }
+}
+
+static MFnAnimCurve::TangentType _ConvertUsdTanTypeToMayaTanType(TsInterpMode usdTanType)
+{
+    switch (usdTanType) {
+    case TsInterpMode::TsInterpHeld: return MFnAnimCurve::TangentType::kTangentStep;
+    case TsInterpMode::TsInterpLinear: return MFnAnimCurve::TangentType::kTangentLinear;
+    default: return MFnAnimCurve::TangentType::kTangentAuto;
+    }
+}
+
+static bool _CreatePlugSpline(
+    MPlug&                    plug,
+    TsSpline                  spline,
+    UsdMayaPrimReaderContext* context,
+    const MDistance::Unit     convertToUnit = MDistance::kMillimeters)
+{
+    auto valueType = spline.GetValueType();
+    if (valueType != Ts_GetType<float>()) {
+        TF_CODING_ERROR(
+            "Unsupported type name for Spline attribute '%s': %s",
+            plug.partialName().asChar(),
+            valueType.GetTypeName().c_str());
+        return false;
+    }
+
+    TsKnotMap knots = spline.GetKnots();
+    if (knots.empty()) {
+        return false;
+    }
+
+    MFnAnimCurve animFn;
+    MStatus      status;
+    MObject      animObj = animFn.create(plug, nullptr, &status);
+    CHECK_MSTATUS_AND_RETURN(status, false)
+
+    unsigned int numKnots = static_cast<unsigned int>(knots.size());
+    MTimeArray   timeArray(numKnots, 0.0);
+    MDoubleArray valuesArray(numKnots, 0.0);
+    MIntArray    tangentInTypeArray(numKnots, 0);
+    MIntArray    tangentOutTypeArray(numKnots, 0);
+    MIntArray    tangentsLockedArray(numKnots, 1);
+    MIntArray    weightsLockedArray(numKnots, 0);
+    MDoubleArray tangentInXArray(numKnots, 0.0);
+    MDoubleArray tangentInYArray(numKnots, 0.0);
+    MDoubleArray tangentOutXArray(numKnots, 0.0);
+    MDoubleArray tangentOutYArray(numKnots, 0.0);
+
+    unsigned int knotIdx = 0;
+    auto         preTanType = MFnAnimCurve::TangentType::kTangentFixed;
+    for (const TsKnot& knot : knots) {
+        float value;
```

Comment:
I'm a bit wary of uninitialzed variables. Maybe default to zero? I know currently all code paths below potentially initialize it. (Although IDK if GetValue() can fail?)